src/storage/src/source/reclock.rs

```diff
@@ -67,12 +68,25 @@ impl ReclockFollower {
                 remap_trace: HashMap::new(),
                 since: as_of,
                 upper: Antichain::from_elem(Timestamp::minimum()),
-                source_upper: HashMap::new(),
+                source_upper: OffsetAntichain::new(),
             })),
         }
     }
 
-    pub fn source_upper(&self) -> Ref<HashMap<PartitionId, MzOffset>> {
+    /// Ensure the `ReclockFollower` has been initialized with trace
+    /// up to the given upper.
+    pub async fn ensure_initialized_to(&self, upper: Antichain<Timestamp>) {
```

Comment:
nit: the arg should probably be an `AntichainRef` since you don't need to own it

Reply:
the ci train has left the station but I will clean up!